### PR TITLE
JSON Response Files And Docs

### PR DESCRIPTION
**Pull request by nicky-johnstone**
Moving over the global JSON response implementation from the blog project to the Qcode-tcl library.

### CODE REVIEW

File: doc/global-json-response.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Details on setting up the response can be found in the [API].
+
+Usage
+-----
+
+### [qc::validate2model]
+This procedure validates input against the data model and modifies the `record` object of the response accordingly.
+
+### Custom Validation Handlers
```
Comment (nicky-johnstone) [inline doc/global-json-response.md]: I actually started somewhat briefly on doing this but stopped to evaluate.

Currently, the registration processes are only for the blog project so adding the registration documentation here doesn't make sense at the moment. I considered moving the `register` and `validate` procs over to the library with this commit but then realised that there's also more stuff that needs moving over and thought it might perhaps be best done one pull request at a time?

Here's what I've established so far that still has to be moved over from blog to the library:

* new handler registration process
* `qc::handlers` API
* new filters
* new `conn_marshal` and the related request handlers
* `cast`, `castable`, and `is` ensembles as well as the `ensemble_handler`
* file management

Should I add the registration process and documentation into this pull request or in a later one?

File: doc/global-json-response.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            "message": ""
+        }
+    },
+    "message": {
+        "alert": {
+            "value": "Sorry, that email or password is not recognised."
+        }
+    },
+    "action": {
+
```
Comment (bvw) [inline doc/global-json-response.md]: +1 for url rather than value. This might need to be a separate pull request in its own right.

File: doc/global-json-response.md (hunk 1 above)
Comment (bvw) [inline doc/global-json-response.md]: That's fine. Smaller chunks is better so I'll merge this one.

**Comment by bvw** [comment]
I think we can move all the work you mentioned into the library.

We need to think about how we upgrade projects for the new cast ensemble and merge existing casts etc.

new handler registration process
qc::handlers API
new filters
new conn_marshal and the related request handlers
cast, castable, and is ensembles as well as the ensemble_handler
file management
